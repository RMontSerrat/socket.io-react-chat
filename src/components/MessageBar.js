import React, { useState } from 'react';
import styled from 'styled-components';
import Input from './Input';
import Button from './Button';

const Container = styled.div`
  display: flex;
`

const ButtonContainer = styled.div`
  width: 250px;
  padding-left: 10px;
`

const MessageBar = ({ onSubmit }) => {
  const [inputValue, changeValue] = useState(null);

  const handleSubmit = () => {
    if (onSubmit) {
      onSubmit(inputValue);
    }
  }

  const handleChange = e => {
    const { target: { value }} = e;
    changeValue(value);
  }

  return (
    <Container>
      <Input
        placeholder="Digite sua mensagem aqui..."
        value={inputValue}
        onChange={handleChange}
      />
      <ButtonContainer>
        <Button onClick={handleSubmit}>Enviar</Button>
      </ButtonContainer>
    </Container>
  )
}

export default MessageBar;