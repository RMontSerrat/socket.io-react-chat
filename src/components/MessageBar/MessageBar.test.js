import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import MessageBar from "./MessageBar";

afterEach(cleanup);
describe('<MessageBar />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <MessageBar />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with onSubmit", () => {
    const onSubmit = jest.fn();
    const { asFragment, getByTestId } = render(
      <MessageBar onSubmit={onSubmit} />
    );
    const input = getByTestId('messageBarInput');
    const message = 'oi pessoal';
    fireEvent.change(input, { target: { value: message }});
    const form = getByTestId('messageBarSubmit');
    fireEvent.submit(form);
    expect(onSubmit).toHaveBeenCalled();
    expect(onSubmit).toHaveBeenCalledWith(message);
    expect(asFragment()).toMatchSnapshot();
  });
});