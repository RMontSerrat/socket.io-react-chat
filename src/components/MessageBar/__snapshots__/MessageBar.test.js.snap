// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageBar /> should render 1`] = `
<DocumentFragment>
  <form
    class="sc-htpNat cixthU"
    data-testid="messageBarSubmit"
  >
    <input
      class="sc-bdVaJa kCgXyL"
      data-testid="messageBarInput"
      placeholder="Digite sua mensagem aqui..."
      value=""
    />
    <div
      class="sc-bxivhb digzrY"
    >
      <button
        class="sc-bwzfXH dMuoyQ"
        type="submit"
      >
        Enviar
      </button>
    </div>
  </form>
</DocumentFragment>
`;

exports[`<MessageBar /> should render with onSubmit 1`] = `
<DocumentFragment>
  <form
    class="sc-htpNat cixthU"
    data-testid="messageBarSubmit"
  >
    <input
      class="sc-bdVaJa kCgXyL"
      data-testid="messageBarInput"
      placeholder="Digite sua mensagem aqui..."
      value=""
    />
    <div
      class="sc-bxivhb digzrY"
    >
      <button
        class="sc-bwzfXH dMuoyQ"
        type="submit"
      >
        Enviar
      </button>
    </div>
  </form>
</DocumentFragment>
`;
